src/classes/class-maintenancemode.php

```diff
@@ -11,78 +11,97 @@
 namespace Niteo\WooCart\Defaults {
 
 	/**
-	 * Class WordPress
+	 * Class MaintenanceMode
 	 *
 	 * @package Niteo\WooCart\Defaults
 	 */
-	class WordPress {
+	class MaintenanceMode {
+
+		public function __construct() {
+			add_action( 'plugins_loaded', array( $this, 'init' ) );
```

Comment:
```suggestion
			add_action( 'template_redirect', array( $this, 'maintenance_mode' ) );
```

We can avoid a lot of bugs this way, as these fires in routing logic the payment gateways will also get the proper response.